main/src/main/scala/sbt/Defaults.scala

```diff
@@ -167,7 +167,17 @@ object Defaults extends BuildCommon {
       fileOutputExcludeFilter :== NothingFilter.toNio,
       inputFileStamper :== sbt.nio.FileStamper.Hash,
       outputFileStamper :== sbt.nio.FileStamper.LastModified,
-      onChangedBuildSource :== sbt.nio.Keys.WarnOnSourceChanges,
+      onChangedBuildSource :== {
+        val sysPropKey = "sbt.build.onchange"
+        sys.props.getOrElse(sysPropKey, "warn") match {
+          case "reload" => sbt.nio.Keys.ReloadOnSourceChanges
+          case "warn"   => sbt.nio.Keys.WarnOnSourceChanges
+          case "ignore" => sbt.nio.Keys.IgnoreSourceChanges
+          case unknown =>
+            System.err.println(s"Unknown $sysPropKey: $unknown.\nUsing warn.")
+            sbt.nio.Keys.WarnOnSourceChanges
+        }
```

Comment:
Thanks for the contribution. Could you move this to https://github.com/sbt/sbt/blob/develop/main/src/main/scala/sbt/internal/SysProp.scala so we can centralize sys props related things there?

Reply:
Thanks for the review. I'll change the code.

Reply:
I've changed it.